src/QsCompiler/Core/Transformations/TypeTransformation.fs

```diff
@@ -123,6 +123,13 @@ type TypeTransformationBase(options: TransformationOptions) =
         if not options.Enable then
             t
         else
+            let range =
```

Comment:
Hm, to make the change in behavior clearer, I would deprecate OnRangeInformation, and instead make separate methods to overload for annotated and inferred ranges. I think that will make it clearer that and how transformations need to be updated. Do you see any downsides to that?

Reply:
I think adding separate abstract methods for each case of `TypeRange` is too much, but I did deprecate `OnRangeInformation` and added a new `OnTypeRange`.

Reply:
Ok, that works for me.